examples/foreach-statement/foreach_statement.md

```diff
@@ -1,7 +1,14 @@
 # Foreach statement
 
-The `foreach` statement iterates over an iterable value, by binding a variable to each member of the iterable value in order. `foreach` also works for strings, and will iterate over each character of the `string`.
+The `foreach` statement iterates over an `iterable` value such as arrays, tuples, maps, records, string, and tables etc. by executing a block of statements for each value in iteration order. For a string value, it will iterate over each code point of the string.
+
+The syntax contains a variable, keyword `in` followed by an expression and a block of statements. The variable can be in the form of a type binding pattern and the expression can be an action invocation, which should evaluate to an iterable value.
```

Comment:
With this "the expression can be an action invocation", a user might think that the expression can only be an action.